synapse/handlers/send_email.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021 The Matrix.org C.I.C. Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import email.utils
+import logging
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from typing import TYPE_CHECKING
+
+from synapse.logging.context import make_deferred_yieldable
+
+if TYPE_CHECKING:
+    from synapse.server import HomeServer
+
+logger = logging.getLogger(__name__)
+
+
+class SendEmailHandler:
```

Comment:
I initially wasn't sure about whether something that didn't immediately relate to processing based off an incoming HTTP request could be considered a `Handler`. But then I looked at `AuthHandler`, which is called from many random places, and realised the definition is definitely not as strict.

Besides that, I noticed that this function is explicitly handling sending Multipart emails. Should we update `send_email`/`SendEmailHandler` to include "Multipart" somewhere?

Reply:
I'm not sure, since Synapse never sends non-multipart emails (and `Mailer` isn't named `MultipartMailer`).